src/dict.c

```diff
@@ -58,6 +58,7 @@
  * between the number of elements and the buckets > dict_force_resize_ratio. */
 static dictResizeEnable dict_can_resize = DICT_RESIZE_ENABLE;
 static unsigned int dict_force_resize_ratio = 5;
+static unsigned int dict_force_shrink_ratio = 10;
```

Comment:
i don't think this is the "force" threshold.
with expand we normally grow up when reaching 100%, unless rehashing is disabled, in which case we grow up only when reaching 500%. (and skip even this threshold if we're in FORBID mode.

what i'd suggest instead is to use the existing HASHTABLE_MIN_FILL (to be used in both htNeedsResize and the new _dictShrinkIfNeeded.
use that threshold with resize is ENABLED, and apply an additional `dict_force_resize_ratio` factor (i.e. 2% instead of 10%) when DISABLED (but not when FORBID)

WDYT?

Reply:
Thanks for your suggestion. I think you are right. We should follow the meaning of `dict_force_resize_ratio `. I have changed it in the latest commit.